src/main/java/org/opensearch/agent/tools/PPLTool.java

```diff
@@ -227,7 +226,7 @@ public <T> void run(Map<String, String> parameters, ActionListener<T> listener)
                         return;
                     }
                     String ppl = parseOutput(dataAsMap.get("response"), indexName);
-                    if (!this.execute) {
+                    if (!clusterSettingHelper.getClusterSettings(SkillSettings.PPL_EXECUTION_ENABLED) || !this.execute) {
                         Map<String, String> ret = ImmutableMap.of("ppl", ppl);
```

Comment:
Can we add a log here when cluster setting and tool setting are conflict? For example, "To execute PPL, please set lugins.skills.ppl_execution_enabled to true". It may be confused if customer set tool setting to true but it is not executed.

Reply:
Sure, will add debug log here.